src/Compiler/Service/ServiceParseTreeWalk.fs

```diff
@@ -521,6 +516,9 @@ module SyntaxTraversal =
                                         visitor.VisitRecordField(path, Some expr, None)
                                     else
                                         None)
+                        | Some(expr, None) ->
+                            // No explicit separator (implicit OBLOCKSEP). Still dive into expr.
+                            yield dive expr expr.Range traverseSynExpr
```

Comment:
Maybe it would make it simpler and less repetitive if it was a nested match, like:
```fsharp
match copyOpt with
| Some(expr, blockSep) ->
    yield dive expr ...

    match blockSep with
    | Some blockSep ->
        ...
```